db/dynamodb-table.yml

```diff
@@ -0,0 +1,149 @@
+# db/dynamodb-table.yml
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >-
+  Create a simple primary-key-lookup DynamoDB table
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Table Definition
+        Parameters:
+          - TableName
+          - PrimaryKey
+          - PrimaryType
+          - ExpirationField
+      - Label:
+          default: Optional Permissions
+        Parameters:
+          - ForeignAccountId
+          - ForeignRoleName
+      - Label:
+          default: Metadata
+        Parameters:
+          - EnvironmentType
+          - ProjectTag
+    ParameterLabels:
+      TableName:
+        default: Table name
+      PrimaryKey:
+        default: Primary key field
+      PrimaryType:
+        default: Primary key type
+      ExpirationField:
+        default: Field to enable expiration on
+      ForeignAccountId:
+        default: Foreign account id
+      ForeignRoleName:
+        default: Role name in foreign account
+      EnvironmentType:
+        default: Environment type
+      ProjectTag:
+        default: Project Tag
+Parameters:
+  TableName:
+    Type: String
+    Description: The unique DynamoDB table name.
+  PrimaryKey:
+    Type: String
+    Description: The name of the primary key field
+  PrimaryType:
+    Type: String
+    Description: The type of the primary key field
+    AllowedValues:
+      - String
+      - Numeric
+      - Binary
+  ExpirationField:
+    Type: String
+    Description: Optionally enable Time-To-Live on this field
+  ForeignAccountId:
+    Type: String
+    Description: Optional account to enable write access to
+  ForeignRoleName:
+    Type: String
+    Description: Optional role name within that account (MUST EXIST)
+  EnvironmentType:
+    Type: String
+    Description: Environment this table is used by.
+    AllowedValues:
+      - Testing
+      - Staging
+      - Production
+  ProjectTag:
+    Type: String
+    Description: The value used for the Project tag on resources that support tagging.
+Conditions:
+  HasExpirationField: !Not [!Equals [!Ref ExpirationField, ""]]
+  HasForeignAccount: !Not [!Equals [!Ref ForeignAccountId, ""]]
+  HasForeignRole: !Not [!Equals [!Ref ForeignRoleName, ""]]
+Mappings:
+  DataTypesMap:
+    String:
+      Symbol: S
+    Numeric:
+      Symbol: N
+    Binary:
+      Symbol: B
+Resources:
+  DynamoDBTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      AttributeDefinitions:
+        - AttributeName: !Ref PrimaryKey
+          AttributeType: !FindInMap [DataTypesMap, !Ref PrimaryType, Symbol]
+      # https://github.com/awslabs/cfn-python-lint/issues/509
```

Comment:
My local `cfn-lint` picks this up.  But apparently the build passed?

Reply:
Also, I went with on-demand pricing for now.  If/when we move enough things to IAB-2.0 that this is expensive, will update to reserved.